File: l-2-2/cs/lib/Triangle.cs
using System;

public class Triangle : Figure
{
    private double ax, ay, bx, by, cx, cy;

    public Triangle(double ax, double ay, double bx, double by, double cx, double cy)
    {
        this.ax = ax; this.ay = ay;
        this.bx = bx; this.by = by;
        this.cx = cx; this.cy = cy;
    }

    private double Distance(double x1, double y1, double x2, double y2)
    {
        return Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
    }

    public override double GetPerimeter()
    {
        return Distance(ax, ay, bx, by) +
               Distance(bx, by, cx, cy) +
               Distance(cx, cy, ax, ay);
    }

    public override double GetArea()
    {
        return Math.Abs(
            (ax * (by - cy) + bx * (cy - ay) + cx * (ay - by)) / 2.0
        );
    }
}

File: l-2-2/ts/derived/Triangle.ts
import { Figure } from "../base/Figure";

export class Triangle extends Figure {
    private ax: number;
    private ay: number;
    private bx: number;
    private by: number;
    private cx: number;
    private cy: number;

    constructor(ax: number, ay: number, bx: number, by: number, cx: number, cy: number) {
        super();
        this.ax = ax; this.ay = ay;
        this.bx = bx; this.by = by;
        this.cx = cx; this.cy = cy;
    }

    private distance(x1: number, y1: number, x2: number, y2: number): number {
        return Math.sqrt((x2 - x1)**2 + (y2 - y1)**2);
    }

    getPerimeter(): number {
        return this.distance(this.ax, this.ay, this.bx, this.by) +
            this.distance(this.bx, this.by, this.cx, this.cy) +
            this.distance(this.cx, this.cy, this.ax, this.ay);
    }

    getArea(): number {
        return Math.abs(
            (this.ax * (this.by - this.cy) + this.bx * (this.cy - this.ay) + this.cx * (this.ay - this.by)) / 2
        );
    }
}
